### PR TITLE
boards: support for the LILYGO TTGO T8 ESP32-S2 board

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the support for the LILYGO TTGO T8 ESP32-S2 board which has a OLED display (not yet supported) and a SD-Card slot on board.

The board is equipped with a USB-C connector that connects either to a USB-to-UART bridge or to the USB-OTG/JTAG interface of the ESP32-S2 via some DIP switches.

The PR includes a very small fix of printf format string in `tests/malloc`. I can split it off.

### Testing procedure

t.b.d.

### CODE REVIEW

**Comment by benpicco** [comment]
Do you know if the display is supported by [u8g2](https://github.com/olikraus/u8g2)?

**Comment by gschorcht** [comment]
> Do you know if the display is supported by [u8g2](https://github.com/olikraus/u8g2)?

The display uses a Sitronix ST7789 which isn't supported by `u8g2`. We have a ST7735 driver in RIOT but it doesn't seem to work with ST77789. I will have a deeper look what the differences are and how it could be supported.

File: boards/esp32s2-lilygo-ttgo-t8/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ifneq (,$(filter mtd,$(USEMODULE)))
+  USEMODULE += mtd_sdcard_default
+endif
+
+# default to using fatfs on SD card
+ifneq (,$(filter vfs_default,$(USEMODULE)))
+  USEMODULE += fatfs_vfs
+  USEMODULE += mtd
+endif
+
```
Comment (benpicco) [inline boards/esp32s2-lilygo-ttgo-t8/Makefile.dep:14]:
```suggestion
ifneq (,$(filter disp_dev,$(USEMODULE)))
  USEMODULE += st7735
endif
```

File: boards/esp32s2-lilygo-ttgo-t8/include/board.h [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#define ST7735_PARAM_CS           GPIO34                /**< Chip Select pin */
+#define ST7735_PARAM_DCX          GPIO37                /**< DCX pin */
+#define ST7735_PARAM_RST          GPIO38                /**< Reset pin */
+#define ST7735_PARAM_RGB          1                     /**< RGB mode enable */
+#define ST7735_PARAM_INVERTED     0                     /**< Inverted mode enable */
+#define ST7735_PARAM_NUM_LINES    135U                  /**< Number of lines */
+#define ST7735_PARAM_RGB_CHANNELS 240U                  /**< Number of columns */
+
+/**< Backlight GPIO initialization */
+#define BACKLIGHT_ON    do { \
```
Comment (benpicco) [inline boards/esp32s2-lilygo-ttgo-t8/include/board.h:103]: No `BACKLIGHT_OFF`?

File: boards/esp32s2-lilygo-ttgo-t8/include/board.h [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                                gpio_init(GPIO33, GPIO_OUT); \
```
Comment (benpicco) [inline boards/esp32s2-lilygo-ttgo-t8/include/board.h:104]: `adafruit-pybadge` just does 

```C
    if (IS_USED(MODULE_ST7735)) {
        gpio_init(BACKLIGHT_PIN, GPIO_OUT);
    }
```

in `board.c`

**Review by benpicco — COMMENTED** [review]
Looks good to me

File: boards/esp32s2-lilygo-ttgo-t8/include/board.h (hunk 2 above)
Comment (gschorcht) [reply boards/esp32s2-lilygo-ttgo-t8/include/board.h:103]: I'm not sure whether `BACKLIGHT_*` is a documented API. I would guess no. `stm32f746g-disco` also defines a `BACKLIGHT_TOGGLE` macro. IMHO, the only macro that is used is `BACKLIGHT_ON`.

File: boards/esp32s2-lilygo-ttgo-t8/include/board.h (hunk 3 above)
Comment (gschorcht) [reply boards/esp32s2-lilygo-ttgo-t8/include/board.h:104]: Hm, this is for initialization. That would also be good for `esp32s2-lilygo-ttgo-t8`. Then we could just define `BACKLIGHT_ON` and `BACKLIGHT_OFF` as `gpio_set` and `gpio_clear`, respectively.

File: boards/esp32s2-lilygo-ttgo-t8/include/board.h (hunk 2 above)
Comment (benpicco) [reply boards/esp32s2-lilygo-ttgo-t8/include/board.h:103]: `disp_dev.h` defines it, but unfortunately does not document it :confused:

File: boards/esp32s2-lilygo-ttgo-t8/include/board.h (hunk 3 above)
Comment (benpicco) [reply boards/esp32s2-lilygo-ttgo-t8/include/board.h:104]: Yes that's the idea

File: boards/esp32s2-lilygo-ttgo-t8/board.c [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * @{
+ *
+ * @file
+ * @brief       Board initialization for the LILYGO TTGO T8 ESP32-S2 board
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#include "board.h"
+#include "periph_gpio.h"
```
Comment (benpicco) [inline boards/esp32s2-lilygo-ttgo-t8/board.c:20]: this file is missing

Comment (benpicco) [inline boards/esp32s2-lilygo-ttgo-t8/board.c:20]:
```suggestion
#include "periph/gpio.h"
```

**Comment by benpicco** [comment]
bors merge

File: boards/esp32s2-lilygo-ttgo-t8/board.c (hunk 4 above)
Comment (gschorcht) [reply boards/esp32s2-lilygo-ttgo-t8/board.c]: Yeah, I squashed it directly.

**Comment by gschorcht** [comment]
Thanks for review.

**Pull request merged**